test/test_dataloader.py

```diff
@@ -2262,10 +2276,31 @@ def test_shuffle(self):
         self.assertNotEqual(items, list(dl))
         self.assertEqual(items, sorted(list(dl)))
 
-        dl = DataLoader2(self.Sorter(dp), batch_size=None, num_workers=2, shuffle=True)
+        dl = DataLoader2(self.IterSorter(dp), batch_size=None, num_workers=2, shuffle=True)
+        self.assertEqual(list(dl), items)
+
+        dl = DataLoader2(self.IterSorter(dp), batch_size=None, num_workers=2, shuffle=True)
+        self.assertEqual(list(dl), items)
+
+    def test_map_shuffle(self):
+        items = list(range(1000))
+        dp = SequenceWrapper(items).shuffle()
+
+        dl = DataLoader2(dp, batch_size=None, num_workers=2, shuffle=False)
+        self.assertEqual(items, list(dl))
```

Comment:
I guess this PR is not to handle the difference between `IterDataPipe.shuffle()` (whose order changes at every epoch) and `MapDataPipe.shuffle()` (whose order do not change)?

Reply:
This PR guarantees those `DataPipes` are shuffled differently by `DataLoader` per epoch

Reply:
Then does that mean `shuffle()` in both IterDataPipe and MapDataPipe will have no effect if `shuffle` argument in `DataLoader` is False?  If so, then why do we still need the `shuffle()` DataPipe?

Reply:
This is because users would be able to choose either shuffle or not with the same pipeline.

Reply:
I see.  Thanks for the clarification!